### DISCUSSION TITLE
bug or feature of npm itself? 'force TLS' button for new proxy host

### DISCUSSION THREAD

**Discussion by shakebox** [ZoeyVid/NPMplus#800] [Q&A] [answered]
Sorry if this is a stupid question. I'm seeing this with all different npm installations i tested, but as a new starter with this topic i wanted to ask advanced users first before maybe filing a bug with the original npm (if that would be the place for it):

whenever i create a new proxy host and have npm create a new cert for it, i also switch on the 'force TLS' option. creation works fine, perfect. then when i go to "edit" of this new host and check the TLS options, the 'force TLS' switch is always off and i manually need to activate it again. is this for you as well? is this on purpose? to me as a newbie it's quite confusing that activating the switch during creation process does not keep it activated after the successful creation. just me?

thx alot!

**Comment by Zoey2936** [comment]
is the setting only missing on new created hosts or also when editing? sounds like a bug

**Comment by shakebox** [comment]
the option is actually called "force HTTPS" in the GUI, not "force TLS". in the created proxy host .conf file its called "Force TLS". Sorry for that confusion.

it's only happening when creating a new (proxy) host for the first time:

- "Add ProxyHost"
- fill in your details and switch to the "TLS" tab
- choose "Request a new TLS certificate", ACTIVATE THE "FORCE HTTPS" SWITCH and finish the creation

when you then go to "edit" for this new host and check the TLS tab, the "force HTTPS" switch is deactivated and in the .conf file the according include line is missing. if you turn the switch on now and save, all is fine and the option is actually active and stays active.

just wondering why no one else ever recognized this. Don't know how many users have proxy host configs in their npm where they assume to always do HTTPS and actually don't because they never re-checked the settings of their configs.

**Reply by Zoey2936** [reply]
I will try to fix it

**Reply by Zoey2936** [reply]
I've tried reproducing it and yes the issue exists, but I found a fix

**Reply by Zoey2936** [reply, marked as answer]
will release a fix today

**Reply by Zoey2936** [reply]
can you please retry with the newest release?

**Reply by shakebox** [reply]
yes, seems to work now. thx. would it make sense to have that reported (and best: fixed) in the original npm? should i file a bug there? it's not an issue of your npmplus but from the original npm, right!?

**Reply by Zoey2936** [reply]
If you want you can do it, the bug should also ecist in upstream because this condition: 
https://github.com/NginxProxyManager/nginx-proxy-manager/blob/79cd0c5294f1e3ad4cc9ed5187b7759808b91162/backend/internal/host.js#L22
Is also triggered if the certificate id is "new"
